test/testNestingDepth.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+from .testHelpers import get_cpp_function_list
+
+
+class TestCppNestingDepth(unittest.TestCase):
+
+    def test_one_function_with_no_nd_condition(self):
+        result = get_cpp_function_list("int fun(){}")
+        self.assertEqual(0, result[0].max_nesting_depth)
+
+    def test_one_function_with_one_nd_condition(self):
+        result = get_cpp_function_list("int fun(){if(a){xx;}}")
+        self.assertEqual(1, result[0].max_nesting_depth)
+
+    def test_one_function_nd_with_question_mark(self):
+        result = get_cpp_function_list("int fun(){return (a)?b:c;}")
+        self.assertEqual(1, result[0].max_nesting_depth)
+
+    def test_one_function_nd_with_forever_loop(self):
```

Comment:
I am not sure if there has been a definitive research in this area,
but I found this http://www.ndepend.com/docs/code-metrics#ILNestingDepth.
It seems to be a suite for Visual Studio.
Their interpretation of the ND seems to be more like local maximum CCN.
For example,
```cpp
if (a && b) {}
```
the statement has ND 2.
If this is the definition of the ND you are implementing,
your test as written looks correct to me.
However, I think this definition of ND doesn't add much value to the existing facilities that calculate CCN.
I suspect that this ND metric has a high correlation with the CCN,
so it seems redundant for me.

The issue #69 is about nesting control structures,
which closely relates to indentation,
so if you need to indent your code deeper than 3 tabs, structures, (i, j, k loops),
your code has problems.
I think this metric or smell detector is more useful in practice.
It is simple and easy to internalize for developers to act upon or refactor the code.